src/nvim/os/shell.c

```diff
@@ -37,6 +37,40 @@ typedef struct {
 #endif
 
 
+/// Process command string with 'shellxescape' (p_sxe) and 'shellxquote'
+/// (p_sxq)
+///
+/// @param cmd Command string
+/// @return A newly allocated command string. It must be freed with
+///         `xfree` when no longer needed.
+static char *cmd_sxe_sxq(const char *cmd)
+{
+  char *ncmd;
+
+  if (cmd == NULL || *p_sxq == NUL) {
+    ncmd = (cmd == NULL) ? NULL : xstrdup(cmd);
+  } else {
+    char *ecmd = (char *)cmd;
+
+    if (*p_sxe != NUL && STRCMP(p_sxq, "(") == 0) {
+      ecmd = (char *)vim_strsave_escaped_ext((char_u *)cmd, p_sxe, '^', false);
+    }
+    ncmd = xmalloc(strlen(ecmd) + STRLEN(p_sxq) * 2 + 1);
+    STRCPY(ncmd, p_sxq);
+    strcat(ncmd, ecmd);
+    /* When 'shellxquote' is ( append ).
+     * When 'shellxquote' is "( append )". */
+    STRCAT(ncmd, STRCMP(p_sxq, "(") == 0 ? (char_u *)")"
+        : STRCMP(p_sxq, "\"(") == 0 ? (char_u *)")\""
+        : p_sxq);
```

Comment:
Might be more readable to split this using `if` instead of using the conditional operator.

Also, since ncmd is a `char` (not `char_u`), I'd suggest using `strcat` and not casting the literals to `char_u`.